smarts/core/tests/test_bubble_hijacking.py

```diff
@@ -120,15 +134,20 @@ def test_bubble_hijacking(smarts, scenarios, bubbles, num_vehicles):
     for _ in range(300):
         smarts.step({})
         for vehicle in index.vehicles:
+            position = Point(vehicle.position)
             for bubble, geometry in zip(bubbles, geometries):
-                position = Point(vehicle.position)
                 in_bubble = position.within(geometry.bubble)
                 is_shadowing = (
                     index.shadow_actor_id_from_vehicle_id(vehicle.id) is not None
                 )
                 is_agent_controlled = vehicle.id in index.agent_vehicle_ids()
 
-                zone_steps = steps_driven_in_zones[bubble.id][vehicle.id]
+                vehicle_id = (
+                    vehicle.id
+                    if traffic_sim == "SUMO"
+                    else re.sub("_\d+$", "", vehicle.id)
+                )
```

Comment:
Yeah, this is related to the problem we discussed on Monday morning.  Specifically, when vehicles are teleported back to the beginning of their routes and given a new chassis.  When Sumo does this, it reassigns the same `vehicle_id` to the new vehicle.  It can get away with this because it "skips a step", where the teleported vehicle is absent from all Provider States and so pybullet tears down its chassis.

But in the `LocalTrafficProvider` implementation, I chose not to "skip a step".  I instead chose to give the new vehicle/chassis a new `vehicle_id`, albeit derived from the original.  (See line 265 and then lines 572-577 of `local_trafffic_provider.py`.)  In my opinion, a new `vehicle_id` for a new chassis is "more correct" than the way Sumo does it. 

But this test (as well as `test_boids.py`) assumed that the `vehicle_id` would not change after teleportation, so it took the short cut of using that as a key for the "zones" counters it maintains.  Lines 145-149 were necessitated by this short cut.  Basically, they strip the suffix added by `bump_id()` such that the `vehicle_id` key to the zones counters will still  be the same after teleportations.

(btw, re your question about where the `traffic_sim` variable is defined, it's on line 119 -- it's a string parameter available to all tests.  However, if you think it should be made explicit, we can add it in the argument list on line 120 as well.)

Having said all of the above, I would agree to  keep the same `vehicle_id` after teleportation if we change things (both Sumo and Smarts) such that instead of creating a new chassis at that point, we keep the existing chassis and just move it to a new location in pybullet (via some method other than the moveTo constraint, which causes the hopscotching I described on Monday).  I have this on my list of things to experiment with...

Reply:
I think that we should be dealing with the vehicle id rather than the chassis. If the vehicle is still persistent but the chassis driving the vehicle id... I think that is a conceptual error.

As to the issue with `traffic_sim`. My understanding of the pytest parameterised arguments is that they have to be provided in the test function argument list or they would be just referenced as a function. I may be wrong.